src/transformers/trainer_tf.py

```diff
@@ -0,0 +1,509 @@
+from .modeling_tf_auto import TFAutoModelForSequenceClassification
+
+
+logger = logging.getLogger(__name__)
+
+
+class TFTrainer(ABC):
+    def __init__(self, **kwargs):
+        """
+        The list of keys in kwargs here should be generic to all the possible models/architectures
+        and not specific to such or such dataset/task.
+        """
+        self.pretrained_model_name_or_path = kwargs.pop("pretrained_model_name_or_path", None)
+        self.optimizer_name = kwargs.pop("optimizer_name", None)
+        self.warmup_steps = kwargs.pop("warmup_steps", None)
+        self.decay_steps = kwargs.pop("decay_steps", None)
+        self.learning_rate = kwargs.pop("learning_rate", None)
+        self.epsilon = kwargs.pop("epsilon", 1e-08)
+        self.loss_name = kwargs.pop("loss_name", None)
+        self.batch_size = kwargs.pop("batch_size", None)
+        self.eval_batch_size = kwargs.pop("eval_batch_size", None)
+        self.distributed = kwargs.pop("distributed", None)
+        self.epochs = kwargs.pop("epochs", None)
+        self.max_grad_norm = kwargs.pop("max_grad_norm", 1.0)
+        self.metric_name = kwargs.pop("metric_name", None)
+        self.max_len = kwargs.pop("max_len", None)
+        self.task = kwargs.pop("task", None)
+        self.datasets = {}
+        self.labels = []
+        self.processor = None
+        self.architecture = None
+
+        # assert len(kwargs) == 0, "unrecognized params passed: %s" % ",".join(kwargs.keys())
+
+    def setup_training(self, checkpoint_path="checkpoints", log_path="logs", cache_dir="cache"):
+        """
+        Setup the different steps to train a model:
+          - check if all the data are given
+          - create the proper strategy
+          - create the features
+          - prepare the model settings
+
+        Args:
+          checkpoint_path: the directory path where the model checkpoints will be saved.
+          log_path: the directory path where the Tensorboard logs will be saved.
+          cache_dir (optional): the directory path where the pretrained model and data will be cached.
+            "./cache" folder by default.
+          data_dir (optional): the directoty path where the data are. This parameter becomes mandatory if
+            the parameters training_data and validation_data are not given.
+          training_data (optional): the training data, either as a list of example or as a tf.data.Dataset.
+            This parameter becomes mandatory if the parameter data_dir is not given.
+          validation_data (optional): the validation data, either as a list of example or as a tf.data.Dataset.
+            This parameter becomes mandatory if the parameter data_dir is not given.
+          test_data (optional): the test data, either as a list of example or as a tf.data.Dataset.
+            This parameter will take the validation data as value if nothing is given.
```

Comment:
Yes I removed them from the signature because for now I focus only when the data are stored in a file and not directly given to the trainer. It was a bit harder to manage both so I decided to go one at a time.